### ISSUE TITLE
`RangesListFactory` parses range operators incorrectly

### ISSUE THREAD

**Issue by sschuberth** [semver4j/semver4j#112]
For example, `RangesListFactory.create("<= 2.6.8 || >= 3.0.0 <= 3.0.1")` results in

```
rangesList = {RangesList@1904} "RangesList[rangesList=[[=2.6.8], [=3.0.0, =3.0.1]]]"
 rangesList = {ArrayList@1909}  size = 2
  0 = {ArrayList@1911}  size = 1
   0 = {Range@1914} "=2.6.8"
    rangeVersion = {Semver@2061} "2.6.8"
    rangeOperator = {Range$RangeOperator@2055} "EQ"
  1 = {ArrayList@1912}  size = 2
   0 = {Range@1917} "=3.0.0"
    rangeVersion = {Semver@2058} "3.0.0"
    rangeOperator = {Range$RangeOperator@2055} "EQ"
   1 = {Range@1918} "=3.0.1"
    rangeVersion = {Semver@2054} "3.0.1"
    rangeOperator = {Range$RangeOperator@2055} "EQ"
```

It seems that operators are dropped / replaced with empty strings that in turn get mapped to the `EQ` operator.

**Comment by piotrooo** [comment]
It's seems to be a problem with spaces after range operator. 
If I modify your example just a little bit I get correct results.

```
RangesList rangesList = RangesListFactory.create("<=2.6.8 || >=3.0.0 <=3.0.1");
RangesList[rangesList=[[<=2.6.8], [>=3.0.0, <=3.0.1]]]
```

**Comment by sschuberth** [comment]
Interesting. I was basically just copying my test string from here (where there are also spaces):

https://github.com/semver4j/semver4j/blob/8a557786039967e3e48030cf55d69c5a0e177dca/src/main/java/org/semver4j/RangesList.java#L14

**Comment by piotrooo** [comment]
Hehe, it's seems to documentation was tricked you :wink:

BTW. Example in documentation is also buggy.

**Comment by piotrooo** [comment]
I'm agree, we should strip any whitespace between the `<=`, `<`, `>=`, `>`.

So range `<= 2.6.8 || >= 3.0.0 <= 3.0.1` should be sanitized to `<=2.6.8 || >=3.0.0 <=3.0.1`.

**Comment by piotrooo** [comment]
@sschuberth fixed. Will be available in 4.1.0.
